x/btcstaking/keeper/power_dist_change.go

```diff
@@ -236,6 +237,17 @@ func (k Keeper) ProcessAllPowerDistUpdateEvents(
 		}
 	}
 
+	// set voting power to 0 if the fp does not have timestamped pub rand
+	for _, fp := range newDc.FinalityProviders {
+		// TODO calling HasTimestampedPubRand potentially iterates
```

Comment:
We could index timestamped pub rand by `fpBtcPK||height` with the return of `0` or `1` for this 
but it would increase our db size a little bit, unless we clean up after passed epoch or something like this

Reply:
Not sure this will eliminate iteration because, for each new epoch, we still need to iterate every `fpBtcPk || height` to update the value

Reply:
once an epoch is better than each begin block ^^